docs/docs/specs/v0/composer.md

```diff
@@ -0,0 +1,190 @@
+# Composer
+
+A developer-first, peer-to-peer knowledge management system.
+
+## Scenarios
+
+1. Users can collaborate on a github issue together
+2. Users can work on folders of markdown files
+3. Developers can extend Composer with custom data sources and visual surfaces
+
+## Extensibility
+
+> "Composer Chrome" refers to the total composer UI organization and all affordances surrounding the content the users interact with - all UI which is not the content itself.
+
+Two levels of complexity are proposed in sequence:
+
+**Static chrome**, where we constrain the application chrome to be statically defined and not extensible by plugins. Elements of the chrome may each expose some API surface for plugins to extend, but the overall elements of the chrome are fixed and can be assumed by all plugins without explicit dependency expressions.
+
+**Dynamic chrome**, where the chrome and it's organization is also extensible and now plugins have to express explicitly the elements of chrome they depend on because their presence can no longer be assumed.
+
+These are in addition to another orthogonal dimension of complexity, the way we load plugins:
+
+**Static plugins**, where a `<Composer />` element is given a list of `Plugin[]` directly as props and has to be compiled into the app at build time.
+
+**Dynamic plugins**, where the app knows how to load plugins via ESM `import` calls in the browser at runtime.
+
+### 1. Static chrome (v1)
+
+Large screens are split vertically into `sidebar` and `content` areas.
+
+```
+.------------.
+|    |     o |
+|    |       |
+|    |       |
+|    |       |
+'------------'
+```
+
+Small screens turn the `sidebar` into a slide-over element that hides by default.
+
+The HALO button (o) is in one of the corners and can be used to access the DXOS Shell.
+
+The sidebar presents a `tree` of nodes which can be populated by plugins.
+
+The `tree` is a logical first solution to the problem of organization and represents a model users and developers are very familiar with and have grown to expect, having found the same model in almost all nearby and competitive products: Notion, Quip, Dropbox Paper, Google Docs, VSCode, Obsidian, ... etc.
+
+Plugins provide components that can fill the content area fully, one at a time, and can sense the state of the tree view (selection, nodes, etc).
+
+Search happens via an overlay [Cmd+K experience](https://cmdk.paco.me/) where a box pops over all the surfaces and presents a fuzzy-search with autocomplete which can be used to locate objects to navigate to, or actions and commands to invoke.
```

Comment:
We should debate modal vs non-modal search. 

1). Many (most) frames will implement some form of search/filtering (inbox, calendar, tasks, etc. will filter individual objects; composer/stacks might filter/highlight matching elements within the AST, etc.) In these cases, search/filtering may be a "semi-sticky" state while work is continued. 

2). Having a consistent (and consistently located) search area (e.g., sidebar) obviates re-implementing (and cluttering) this functionality in all of these places.

3). When real estate exists (i.e., other than phone) it would be helpful for search not to occlude the main content.

4). Universal search across spaces should have a high priority.

5). Google-style text "search" will likely soon become obsolete. We should consider a hybrid semi-interactive search/suggest model that is "always on" and always visisble.

Reply:
replaced the cmd+k search with universal search description and mechanism in `AppState`